sim/pokemon.js

```diff
@@ -611,12 +611,18 @@ class Pokemon {
 	gotAttacked(move, damage, source) {
 		if (!damage) damage = 0;
 		move = this.battle.getMove(move);
-		this.lastAttackedBy = {
-			pokemon: source,
+		let lastHurtBy = {
+			source: source,
 			damage: damage,
 			move: move.id,
 			thisTurn: true,
 		};
+		this.hurtBy.push(lastHurtBy);
+	}
+
+	get lastHurtBy() {
```

Comment:
I don't like getters for various reasons, the main one being readability – it's not obvious from a glance that this property is automatically updated rather than something you need to write to.